packages/location-management/src/ducks/locationHierarchy/utils.tsx

```diff
@@ -73,3 +73,53 @@ export const generateJurisdictionTree = (apiResponse: RawOpenSRPHierarchy): Tree
 export const serializeTree = (trees: TreeNode[]) => {
   return JSON.stringify(trees.map((tree) => JSON.stringify(cycle.decycle(tree))));
 };
+
+/**
+ * Find the specifc Hierarchy node inside HierarchyTree
+ *
+ * @param {ParsedHierarchyNode} hierarchy HeirarchyTree to search in
+ * @param {string} id Id of the node to search for
+ * @returns {ParsedHierarchyNode | undefined} returns the Node if found else undefined
+ */
+export function getHierarchyNode(
+  hierarchy: ParsedHierarchyNode,
+  id: string
+): ParsedHierarchyNode | undefined {
+  // convert the ParsedHierarchyNode object into tree
+  const tree = new TreeModel().parse(hierarchy);
+  // variable to store either we found node in the tree or not
+  let result: ParsedHierarchyNode | undefined = undefined;
```

Comment:
the parsed hierarchy node is not in the form of Treenode and the data flow this pr implements is of ParsedHierarchyNode not Treenode and over here instead of creating a another recursion function i just used treeModel to walk through the treenodes

Reply:
what is the difference between ParsedHierarchyNode and Treenode?, I thought one was just an abstraction of the other.

Reply:
Treenode is a treemodel object but the ParsedHierarchyNode is a JSON object

Reply:
@ashfahan oh, ok. thanks for that correction.